stdlib/kokorawr/kokorawr/functions/services/ttt.js

```diff
@@ -43,70 +39,89 @@ function isTie(marked, gameOver) {
 * @returns {object} {text: str, success: bool}
 */
 module.exports = (team, loc, reset=false, turn=false, state=false, context, callback) => {
-	if(turn) {
-	    let color = 'Red';
-	    if(lastTeam == 0) {
-	      color = 'Blue';
-	    }
-	    return callback(null, {text: `It is ${color}'s turn!`, success: true});
-  	} else if(state) {
-  		return callback(null, {text: '\n'+markedToBoard(marked), success: true});
-  	} else if(reset && gameOver) {
-		lastTeam = Math.round(Math.random());
-		gameOver = false;
-		marked = ['~', '~', '~', '~', '~', '~', '~', '~', '~'];
-		let color = 'Red';
-	    if(lastTeam == 0) {
-	      color = 'Blue';
-	    }
-	    tracker(true, {'name': 'ttt', 'info': marked}, undefined, function (err, result) {
-			return callback(null, {text: `Successfully Reset! It is ${color}'s turn!`, success: true});
-		});
-	} else if(gameOver) {
-		return callback(null, {text: 'Game already over\n' + markedToBoard(marked), success: false})
-	} else if(reset) {
-		return callback(null, {text: "Can't reset: game not over\n" + markedToBoard(marked), success: false})
-	} else if(lastTeam == team) {
-		let color = 'Red';
-	    if(team != 0) {
-	      color = 'Blue';
-	    }
-		return callback(null, {text: `It is not ${color}'s turn.`, success: false})
-	} else if (marked[loc-1] != '~'){
-		return callback(null, {text: `Location ${loc} already marked!\n` + markedToBoard(marked), success: false});
-	} else {
-		marked[loc-1] = team;
-		lastTeam = team;
+	lib.utils.storage.get('ttt', (err, ttt) => {
+		if (err) {
+			return callback(null, 'An error has occurred with your command');
+		}
+		if (ttt == null) {
+			ttt = {};
+		}
+		if (!ttt.hasOwnProperty('marked')) {
+			ttt['marked'] = ['~', '~', '~', '~', '~', '~', '~', '~', '~'];
```

Comment:
Maybe we should make a constant variable called EMPTY and set it equal to '~' and replace all of the instances of it in the code. It'll make it much more readable.

Reply:
I'll change them for all the files then.

Reply:
yay for cleaner code!